tests/types/test_schema.py

```diff
@@ -1636,6 +1634,59 @@ def test_schema_inference_on_lists(data, data_type):
     assert inferred_schema == expected_schema
 
 
+def test_schema_inference_with_empty_lists():
+    # If (non-nested) empty list is passed, should be inferred as string for backward compatibility
+    # with version 2.7.1: https://github.com/mlflow/mlflow/blob/v2.7.1/mlflow/types/utils.py#L150
+    data = []
+    assert _infer_schema(data) == Schema([ColSpec(DataType.string)])
+
+    # Nested list contains only an empty list is not allowed.
+    data = [[]]
+    with pytest.raises(
+        MlflowException,
+        match=r"A column of nested array type must include at least one non-empty array.",
+    ):
+        _infer_schema(data)
+
+    # If at least one of sublists is not empty, we can assume other empty lists have the same type.
+    data = [
+        {
+            "data": [
+                ["a", "b", "c"],
+                [],
+                ["d", "e"],
+                [],
+            ]
+        }
+    ]
+    inferred_schema = _infer_schema(data)
+    expected_schema = Schema([ColSpec(Array(Array(DataType.string)), name="data")])
+    assert inferred_schema == expected_schema
```

Comment:
Could we add cases for pandas/spark DataFrame as well?

Reply:
Well, pandas are converted to list beforehand so it should be covered already.  Spark seems to have different flow (`_infer_spark_type` func). It should be worth to check pandas/pyspark tests overall, but can I do that in separate PR to avoid scope creep:p